### PR TITLE
Update nddata docs

### LINKED ISSUE #4756: Emphasize NDDataArray in user docs?
I think that most users (as opposed to developers) who are looking in the NDData documentation are actually looking for the `NDDataArray` class.  The current `NDData` object and docs seem largely aimed at developers instead of typical users.  

As someone who hasn't worried much about `NDData` (for lack of personal need), I was surprised the Getting Started docs do not start with a simple _useful_ example like reading a generic FITS image (with or without WCS), plotting it, modifying a keyword, modifying some pixel values, and writing back to file.  In reading the docs I initially got the impression that the only way to do this was to define my own class with the appropriate mixins.  There is just one reference to `NDDataArray` in the docs, offered almost as an aside.  

Part of this is to continue pushing the idea of the unified I/O interface where the data file format (e.g. FITS or HDF5) is not emphasized so much as the data object (Table or NDData).

So what about starting the Getting Started with a section which is the most common use case that utilizes `NDDataArray`, and then perhaps using `NDDataArray` in more of the subsequent examples where it might make sense.  Or perhaps explicitly split into Basic Usage and Developer Usage sections?

### PR DESCRIPTION
**Pull request by mwcraig**
This is an initial draft of the rewrite. I've skipped CI for now because this doesn't need to get merged before feature freeze.

It is written assuming that #6173 (adds the `CCDData` class) will be merged.

~~A rendered copy is at: http://physics.mnstate.edu/craig/apy20/nddata/index.html~~

Edit: closes #4756

### CODE REVIEW

**Comment by bsipocz** [comment]
@mwcraig - ``[docs only]`` now really works for pull requests.

File: docs/nddata/index.rst [hunk 1]
```diff
@@ -7,138 +7,258 @@ N-dimensional datasets (`astropy.nddata`)
...
+    data[100, 300:310] = cosmic_ray_value
+    plt.imshow(data, origin='lower')
+
+
+The `~astropy.nddata.CCDData` object, like the other objects in this package,
+can store the data, a mask, and metadata. The `~astropy.nddata.CCDData` object
+requires that a unit be specified::
+
+    >>> mask = (data == cosmic_ray_value)
+    >>> from ccdproc import CCDData   # doctest: +SKIP
```
Comment (bsipocz) [inline docs/nddata/index.rst]: ``ccdproc`` needs to be changed here

File: docs/nddata/index.rst [hunk 2]
```diff
@@ -7,138 +7,258 @@ N-dimensional datasets (`astropy.nddata`)
...
+    >>> position = (149.7, 100.1)
+    >>> size = (80, 100)     # pixels
+    >>> cutout = Cutout2D(ccd, position, size)
+    >>> plt.imshow(cutout.data, origin='lower')
+
+.. plot::
+
+    import matplotlib.pyplot as plt
+    from astropy.modeling.models import Gaussian2D
+    from ccdproc import CCDData
```
Comment (bsipocz) [inline docs/nddata/index.rst]: remove ``ccdproc``

File: docs/nddata/index.rst [hunk 3]
```diff
@@ -7,138 +7,258 @@ N-dimensional datasets (`astropy.nddata`)
...
+This cutout can also plot itself on the original image::
+
+    >>> plt.imshow(ccd, origin='lower')
+    >>> cutout.plot_on_original(color='white')
+
+.. plot::
+
+    import matplotlib.pyplot as plt
+    from astropy.modeling.models import Gaussian2D
+    from ccdproc import CCDData
```
Comment (bsipocz) [inline docs/nddata/index.rst]: ditto

File: docs/nddata/index.rst [hunk 4]
```diff
@@ -7,138 +7,258 @@ N-dimensional datasets (`astropy.nddata`)
...
+    >>> type(smaller)
+    numpy.ndarray
+    >>> plt.imshow(smaller, origin='lower')
+
+.. plot::
+
+    import matplotlib.pyplot as plt
+    from astropy.modeling.models import Gaussian2D
+    from astropy.nddata import block_reduce, block_replicate
+    from ccdproc import CCDData
```
Comment (bsipocz) [inline docs/nddata/index.rst]: ditto

**Comment by bsipocz** [comment]
@mwcraig - Could you please wrap this up, so I can include it in the RC?

**Comment by mwcraig** [comment]
Will do in the next three hours, @bsipocz

**Comment by mwcraig** [comment]
@bsipocz -- the failure on circle is because the functions in the new doctests require `skimage`. Should I `doctest: +SKIP` those?

Not sure why circle ran on a docs-only buidl either....

**Comment by bsipocz** [comment]
@mwcraig - neither circle nor appveyor understands [docs only]. also doctests run with every run, so those should ideally have either a skip or a doctest requires clause (similar to the ones we do for scipy)

**Comment by mwcraig** [comment]
The answer to the latest fail is, ironically, at: https://stackoverflow.com/questions/24504301/what-does-title-level-inconsistent-mean

**Comment by mwcraig** [comment]
@bsipocz -- merge at will, assuming this (finally) passes. May flesh out a bit more tonight if I have time, but nothing critical.

**Comment by mwcraig** [comment]
@crawfordsm @MSeifert04 @larrybradley -- comments/suggestions welcome. Cutouts and other image utilities are much more prominently featured now.

**Comment by bsipocz** [comment]
Merging this now for the RC. Please open a new PR if there is still more docs to be added for the actual release.

**Pull request merged**